### ISSUE TITLE
JetEngine Listing - add access to fields from "SQL query" in elementor widgets

### ISSUE THREAD

**Issue by LevelUpGit** [CrocoBlock/suggestions#5430]
Plugin: JetEngine
Options in use: Listing, Query builder

Issue: Can't access fields from "SQL query" in Elementor widgets, like we can in dynamic field widget, when creating listing.

Steps:
1. Create SQL query on custom table
2. Create listing for  SQL query
3. Create a new "Heading" and try to pull a dynamic field from an SQL query.

Solution: Add access to fields in Elementor widget, as well in "make listing clickable" option, "Dynamic image", "Dynamic link" widgets.

![1](https://user-images.githubusercontent.com/69863985/171015476-278a08ce-105c-42c7-b57e-59641b19e3fb.jpg)
![2](https://user-images.githubusercontent.com/69863985/171015439-3358ba93-c71b-447b-b208-6519fab47d81.jpg)
![Screenshot_4](https://user-images.githubusercontent.com/69863985/171015492-d3c9b104-68b5-49b2-9e43-7ae9f0b45096.jpg)

**Comment by KidDisco** [comment]
Agreed. This is a pretty big oversight. As far as I can tell, access to this data is only available in the DynamicField widget. This makes the Dynamic Link, Dynamic Image etc, kind of pointless. 

I'm trying to create a DynamicTable with a link. However, the Dynamic Field widget cannot specify *BOTH* the output and a link (nor can I construct a link). At the same time, the DynamicLink widget does not have access to the underlying query created to create a listing or DynamicTable.

In general, if we are to create dynamic sites and leverage the QueryBuilder, access to the underlying query should be made available to all widgets.
